### PR TITLE
Home page display

### PR DESCRIPTION
**Pull request by PrettyPearl2302**
## Description
I was able to successfully display podcasts from my API (that i created) on my home page. I created a route "/api/home" that used the function searchForTerm to be able to get a json of podcasts by genre. essentially, i created a dictionary, so i also wrote the code on the frontend to be able to go through the diary and display the podcasts on the home page by genre. very challenging, but i got it done. woohoo. 

one thing that i still need to accomplish to tidy the headers of each section to be more precise (e.g Business as opposed to PODCASTSERIES_BUSINESS). I also want to enable a horizontal scroll across each genre heading. 

## Milestones
Building home page

## Resources
my student store code from second week of codepath

## Test Plan

Screenshot 2023-07-14 at 12 14 16 AM

### CODE REVIEW

File: timify-api/routes.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+app.get('/api/home', async (req, res) => {
+  const term = ""; 
+  const page = 1;
+  const limitPerPage = 10;
+  const Genres = ["PODCASTSERIES_BUSINESS", "PODCASTSERIES_COMEDY", "PODCASTSERIES_HEALTH_AND_FITNESS", "PODCASTSERIES_NEWS", "PODCASTSERIES_SOCIETY_AND_CULTURE", "PODCASTSERIES_SPORTS"]
+
+  let podcastByGenre = {};
+  let errorOccurred = null;
+
+  for (let Genre of Genres) {
```
Comment (ssorenson130) [inline timify-api/routes.js]: I think this can be `for (const Genre of Genres)`

File: timify-api/routes.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      try {
+          const podcasts = await searchForTerm(term, page, limitPerPage, [Genre]);
+          // console.log(Genre, podcasts)
```
Comment (ssorenson130) [inline timify-api/routes.js]: delete

File: timify-api/seedDatabase.js [hunk 3]
```diff
@@ -1,27 +1,32 @@
-import fs from 'fs';
-import path from 'path';
-import { fileURLToPath } from 'url';
-import { Podcast } from './models/index.js';
+// import fs from 'fs';
+// import path from 'path';
+// import { fileURLToPath } from 'url';
```
Comment (ssorenson130) [inline timify-api/seedDatabase.js]: same here - delete or uncomment

File: timify-api/seedDatabase.js [hunk 4]
```diff
@@ -1,27 +1,32 @@
...
+//     await Podcast.bulkCreate(podcastData);
+//     console.log('Podcast data has been seeded!');
+//   } catch (error) {
+//     console.error('Error seeding data:', error);
+//   } finally {
+//     await sequelize.close();
+//   }
+// };
+
+// seedDatabase();
```
Comment (ssorenson130) [inline timify-api/seedDatabase.js]: same here

**Review by ushanakg — APPROVED** [review]

File: timify-ui/src/components/PodcastGrid/PodcastGrid.jsx [hunk 5]
```diff
@@ -3,19 +3,24 @@ import { AiOutlineLoading } from "react-icons/ai"
...
-        <h3 className='podcast-header'>Podcasts</h3>
-        <div className="podcast-grid">
-            { podcasts.map((podcast, i) => (
-                    <PodcastCard podcast={podcast} key={i} />
-                )
-            )}
+    <div className="podcastGrid">
+      {podcastsByGenre && Object.entries(podcastsByGenre).map(([genre, podcasts]) => (
+        <div key={genre} className="genre-section">
+          <h2>{genre}</h2>
```
Comment (ghost) [inline timify-ui/src/components/PodcastGrid/PodcastGrid.jsx:12]: You may want to create a dict to map each genre param string to a readable string and use it here

Comment (PrettyPearl2302) [reply timify-ui/src/components/PodcastGrid/PodcastGrid.jsx:12]: i was actually wondering how i would achieve that. thank you for this suggestion!

**Pull request merged**